File: task-03/C/Subtask4.c
#include <stdio.h>

int main() {
    FILE *f = fopen("input.txt", "r");
    int n;
    fscanf(f, "%d", &n);
    fclose(f);

    f = fopen("output.txt", "w");
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n-i-1; j++) fprintf(f, " ");
        for (int j = 0; j < 2*i+1; j++) fprintf(f, "*");
        fprintf(f, "\n");
    }
    for (int i = n-2; i >= 0; i--) {
        for (int j = 0; j < n-i-1; j++) fprintf(f, " ");
        for (int j = 0; j < 2*i+1; j++) fprintf(f, "*");
        fprintf(f, "\n");
    }
    fclose(f);
    return 0;
}

File: task-03/Rust/Subtask4.rs
use std::fs;
use std::io;

fn main() -> io::Result<()> {
    let n: usize = fs::read_to_string("input.txt")?.trim().parse().unwrap();
    let mut diamond = String::new();

    for i in 0..n {
        diamond.push_str(&" ".repeat(n - i - 1));
        diamond.push_str(&"*".repeat(2 * i + 1));
        diamond.push('\n');
    }

    for i in (0..n-1).rev() {
        diamond.push_str(&" ".repeat(n - i - 1));
        diamond.push_str(&"*".repeat(2 * i + 1));
        diamond.push('\n');
    }

    fs::write("output.txt", diamond)?;
    Ok(())
}

